src/main/java/org/takes/facets/forward/TkForward.java

```diff
@@ -107,21 +111,23 @@ public InputStream body() throws IOException {
          * @throws IOException If fails
          */
         private Response load() throws IOException {
-            synchronized (this.saved) {
-                if (this.saved.isEmpty()) {
-                    Iterable<String> head;
-                    InputStream body;
-                    try {
-                        head = this.origin.head();
-                        body = this.origin.body();
-                    } catch (final RsForward ex) {
-                        head = ex.head();
-                        body = ex.body();
+            if (this.saved.get() == null) {
+                synchronized (this.mutex) {
+                    if (this.saved.get() == null) {
```

Comment:
@essobedo doesn't it smell for you? looks rather strange. what's wrong about `synchronized` block we had before? why do we need to refactor it?